src/mesh_connectivity.cpp

```diff
@@ -476,3 +479,40 @@ std::vector<std::vector<bool>> sign_propagation(const std::vector<std::vector<si
     }
     return cell_function_label;
 }
+
+std::vector<size_t> sign_propagation_MI(const std::vector<std::vector<size_t>>& material_cells,
+                      const std::vector<size_t>& shell_of_half_patch,
+                      const std::vector<std::vector<size_t>>& shells,
+                      const std::vector<std::pair<size_t, size_t>>& patch_function_label,
+                      size_t n_func,
+                      const std::vector<double>& sample_function_label){
+    std::vector<size_t> cell_function_label(material_cells.size(), Mesh_None);
+    for (size_t cell_index = 0; cell_index < material_cells.size(); cell_index++){
+        for (auto shell : material_cells[cell_index]){
+            /// resolve degenerate cases where all materials are dominated, i.e., the cell contains `Mesh_None` as its element.
```

Comment:
"all materials are dominated" is not an accurate description of this degenerate case.